modules/core/src/main/java/org/apache/ignite/internal/client/thin/TcpClientChannel.java

```diff
@@ -109,6 +131,10 @@
 
             out = sock.getOutputStream();
             in = sock.getInputStream();
+
+            GridUnsafeDataInput dis = new GridUnsafeDataInput();
```

Comment:
What are benefits over standard `DataInputStream`?

Reply:
`DataInputStream` is a big-endian input stream, while the protocol implies little-endian.

Reply:
Did not know it, thank you!